cli/tools/doc.rs

```diff
@@ -146,6 +146,18 @@ pub async fn doc(flags: Flags, doc_flags: DocFlags) -> Result<(), AnyError> {
       if doc_flags.lint {
         let diagnostics = doc_parser.take_diagnostics();
         check_diagnostics(&diagnostics)?;
+        if doc_flags.html.is_none() && !doc_flags.json {
```

Comment:
Just realized: This condition can be removed and instead check for the lint flag when going to output.